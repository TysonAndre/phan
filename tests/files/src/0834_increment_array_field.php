<?php

function test834(array $foo) {
    $resultArr = [
        'requests' => 0,
        'responsetime' => 123,
    ];

    // some foreach loop
    foreach ($foo as $_) {
        $resultArr['requests']++;
    }

    if ($resultArr["requests"] > 0) {  // should not emit PhanDivisionByZero
        $resultArr["responsetime"]    = 2 / $resultArr["requests"];
    }
}

function test834b() {
    $resultArr = [
        'requests' => 0,
        'foo' => 0,
    ];
    $resultArr['requests']++;
    --$resultArr['foo'];
    return $resultArr['requests'] > $resultArr['foo'];
}
