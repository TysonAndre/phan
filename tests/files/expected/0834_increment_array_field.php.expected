%s:15 PhanUnusedVariable Unused definition of variable $resultArr
%s:26 PhanSuspiciousValueComparison Suspicious attempt to compare $resultArr['requests'] of type 1 to $resultArr['foo'] of type -1 with operator '>'
